ipaserver/plugins/service.py

```diff
@@ -289,7 +289,14 @@ def check_required_principal(ldap, principal):
     except errors.ValidationError:
         service_types = ['HTTP', 'ldap', 'DNS', 'dogtagldap']
         if principal.service_name in service_types:
-            raise errors.ValidationError(name='principal', error=_('This principal is required by the IPA master'))
+            raise errors.ValidationError(
+                name='principal',
+                error=_(
+                    '{} is required by the IPA master'.format(
```

Comment:
You have to call format on the i18n translation string, not on the string. Otherwise translation won't work correctly:

``_('{} is required by the IPA master').format(principal)``